### ISSUE TITLE
VGG_CAM weight creation

### ISSUE THREAD

**Issue by 1991viet** [imatge-upc/retrieval-2017-cam#2]
Dear authors,
I have a question regarding the CAM-VGG-16 Weights you provided. Could you tell me how did you have this file. As I guess, you first built the classic VGG model, then remove the fully connected layer by a pair of Conv2d + GAP layers (as you described). Then you load the pretrained VGG weights (caffemodel or sth like that) into layers from the beginning until the relu5_3 layers, and randomly initialize weights for all layers after relu5_3. If I am correct, could you tell me more detail on the whole process ? Specifically, could you share me the repository you use to train you VGG_CAM model ? I found out this one:
https://github.com/tdeboissiere/VGG16CAM-keras/blob/master/VGGCAM-keras.py
which would be very similar to the VGG_CAM model you propose, except from the GAP layer. But I want to mimic the process you train as close as possible, so it would be great if you could share your code. In case it is impossible, please tell me parameters (learning rate, momentum, decay,  number of epochs, minibatch size) and optimizer you use to train your model. And, which dataset you use to train your VGG_CAM model (ILSVRC12 or ILSVRC14 classification dataset or anything else).
For the purpose of this post, I am doing a project which is based on your model. I want to improve it by modifying the VGG_CAM part in different ways (for example, change the GAP layer). But before that, I want to re-produce the same trained model (produce something as similar to CAM-VGG-16 Weights you provided as possible). Then it would be fair to compare performance between your model and my modified model.
Thank you in advance.

**Comment by jsalbert** [comment]
We converted the model from the original [source](https://github.com/metalbubble/CAM) that was in caffe to Keras. If I recall correctly we used [this](https://github.com/MarcBS/keras) We did not train a new VGG16-CAM model. The modifications were done on the converted model and the weights resaved again in Keras format.  
As you said if you want to train a new CAM model, the easiest way would be loading a pre-trained model, removing the top layers, adding the extra conv + GAP layers and fine-tune for some epochs, with a small learning rate.
